### PR TITLE
Fix cancelNext not rescheduling when using cron expressions

### LINKED ISSUE #341: CancelNext does not reschedule later invocations when using a cron expression
I was expecting the following code to cancel one invocation but it is actually cancelling them all : 
```javascript
let schedule = require('node-schedule');

var j = schedule.scheduleJob('*/1 * * * * *', function(){
  console.log('Job every 1 seconds');
});

console.log(j.pendingInvocations().length) // prints 1
console.log(j.nextInvocation().toString()) //prints date in 1 sec

j.cancelNext(true);

console.log(j.pendingInvocations().length) //prints 0
console.log(j.nextInvocation().toString()) //error as nextInvocation is null
```

The cancelNext function is looking for a recurs field on the recurrenceRule which is not present when using a cron expression :
```javascript
this.cancelNext = function(reschedule) {
...
if (reschedule && nextInv.recurrenceRule.recurs) {
      newInv = scheduleNextRecurrence(nextInv.recurrenceRule, this, nextInv.fireDate, nextInv.endDate);
      if (newInv !== null) {
        this.trackInvocation(newInv);
      }
 }
...
}
```
Is this intentional and means you can only cancel jobs when using the RecurrenceRules API ?

### PR DESCRIPTION
**Pull request by joshmeads**
Fixes #341 

Patch provided by @santigimeno

### CODE REVIEW

File: lib/schedule.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ function Job(name, job, callback) {
 
       cancelInvocation(inv);
 
-      if (reschedule && inv.recurrenceRule.recurs) {
+      if (reschedule && inv.recurrenceRule.recurs || nextInv.recurrenceRule.next) {
```
Comment (santigimeno) [inline lib/schedule.js]: The OR should be grouped for this to work. See original patch

File: lib/schedule.js [hunk 2]
```diff
@@ -125,7 +125,7 @@ function Job(name, job, callback) {
 
     cancelInvocation(nextInv);
 
-    if (reschedule && nextInv.recurrenceRule.recurs) {
+    if (reschedule && nextInv.recurrenceRule.recurs || nextInv.recurrenceRule.next) {
```
Comment (santigimeno) [inline lib/schedule.js]: Same here

**Comment by joshmeads** [comment]
@santigimeno please review

**Comment by santigimeno** [comment]
Landed in https://github.com/node-schedule/node-schedule/commit/58b8ec903912f43de5575d147b8dcf0375d34849. Thanks!

**Comment by joshmeads** [comment]
@santigimeno do we know when node-schedule will do a patch release?

**Comment by santigimeno** [comment]
Published in `v1.3.2`

**Pull request closed without merging**